query-engine/js-connectors/smoke-test-js/src/connector/neon.ts

```diff
@@ -110,8 +105,35 @@ class PrismaNeon implements Connector, Closeable {
    * example. The version string is returned directly without any form of
    * parsing or normalization.
    */
-  version(): Promise<string | undefined> {
-    return Promise.resolve(this.maybeVersion)
+  async version(): Promise<string | undefined> {
+    if (this._version) {
+      return Promise.resolve(this._version)
```

Comment:
maybe

```suggestion
      return this._version
```

here and below where applicable since it's already an async function